src/Humanizer/ArticlePrefixSort.cs

```diff
@@ -48,60 +48,68 @@ public static string[] PrependArticleSuffix(string[] appended)
         {
             var inserted = new string[appended.Length];
 
+            // ReSharper disable InconsistentNaming
+            var _The = " The".AsSpan();
+            var The = _The[1..];
+            var _A = " A".AsSpan();
+            var A = _A[1..];
+            var _An = " An".AsSpan();
+            var An = _An[1..];
+            var _a = " a".AsSpan();
+            var a = _a[1..];
+            var _an = " an".AsSpan();
+            var an = _an[1..];
+            var _the = " the".AsSpan();
+            var the = _the[1..];
+            // ReSharper restore InconsistentNaming
             for (var i = 0; i < appended.Length; i++)
             {
-                string suffix;
-                string original;
-                var append = appended[i];
-                if (append.EndsWith(EnglishArticles.The.ToString()))
+                var append = appended[i].AsSpan();
+                if (append.EndsWith(The))
                 {
-                    suffix = append.Substring(append.IndexOf(" The", StringComparison.CurrentCulture));
-                    original = ToOriginalFormat(appended, suffix, i);
-                    inserted[i] = original;
+                    var suffix = append[append.IndexOf(_The, StringComparison.CurrentCulture)..];
```

Comment:
1. `suffix` here is known to be `" The"`. It does not make sense to try to extract it from the string.
2. `append.IndexOf(_The, StringComparison.CurrentCulture)` should be `append.LastIndexOf(_The, StringComparison.CurrentCulture)`. Which is known to be `append.Length - " The".Length`
3. `ToOriginalFormat(append, suffix)` again does the same operation `append.IndexOf(suffix, StringComparison.CurrentCulture)`